src/bootstrap/compile.rs

```diff
@@ -678,13 +678,9 @@ impl Step for Rustc {
                 false
             }
         } else if let Some(path) = &builder.config.rust_profile_use {
-            if compiler.stage == 1 {
```

Comment:
why was this condition removed? Shouldn't it be the same as above, `!link_shared() || stage >= 2`?

Reply:
If a profile is provided there is no harm in using it.  If there are checksum missmatches or missing functions then we have the same result as not using one, but if there are signature matches then the Stage1 compiler will be faster and the total build should complete sooner.